gradle/local-distribution.gradle

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * This script sets up a local distribution including.
+ * To install a local distribution run `installLocalDistribution`.
+ * The local distribution will be installed to
+ * build/distributions/local
+ * */
+import org.elasticsearch.gradle.Architecture
+
+apply plugin:'elasticsearch.distribution-download'
+
+elasticsearch_distributions {
+  local {
+    flavor = 'default'
+    type = 'archive'
+    architecture = Architecture.current()
+  }
+}
+
+tasks.register('installLocalDistribution', Copy) {
+  from(elasticsearch_distributions.local.extracted)
+  into("build/distribution/local")
+}
```

Comment:
I think we should add a println that states the location of the expanded distribution

Reply:
Good call, let's add a user friendly `doLast` which prints something like `Elasticsearch distribution installed to {location}`.

Reply:
done.